ONSdigital/rm-survey-service
# Add create survey classifier endpoint

# Motivation and Context

For a survey to be usable it needs classifiers, this PR adds an endpoint to post classifiers to a given survey.

# What has changed



* Added `POST /surveys/{surveyId}/classifiers` endpoint
* Added unit tests for endpoint
* Added integration test for post classifiers and info endpoint
* Added make target to run the service in docker and run integration tests against it
* Updated API.md and README.md according to changes
* Changed .travis.yml to run integration tests instead of unit tests and docker as these are all run as part of the integration test target

# How to test?



Check the tests are valid and cover the requirements.

Run `make integration-test` target to make sure it is spinning up the services, running the tests and tearing down what it runs appropriately.

Try inserting `So(false, ShouldBeTrue)` into an integration test to induce a failure then run `make integration-test` and observe the tests should fail, container teardown still happens and the target exits with an error code.

# Links



https://trello.com/c/cnqv2QuY/182-add-endpoint-in-survey-service-to-add-classifiers-8
